web-api/src/embedding.rs

```diff
@@ -17,14 +17,26 @@ use xayn_ai_bert::{AvgEmbedder, Config as EmbedderConfig, NormalizedEmbedding};
 
 use crate::{app::SetupError, error::common::InternalError, utils::RelativePathBuf};
 
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "lowercase", deny_unknown_fields)]
```

Comment:
🔧 for correct formatting if we add variants with more than one word in the future
```suggestion
#[serde(tag = "type", rename_all = "snake_case", deny_unknown_fields)]
```